src/reflect/value.go

```diff
@@ -794,7 +890,30 @@ func (v Value) Convert(t Type) Value {
 }
 
 func MakeSlice(typ Type, len, cap int) Value {
-	panic("unimplemented: reflect.MakeSlice()")
+	if typ.Kind() != Slice {
+		panic("reflect.MakeSlice of non-slice type")
+	}
+	if len < 0 {
+		panic("reflect.MakeSlice: negative len")
+	}
+	if cap < 0 {
+		panic("reflect.MakeSlice: negative cap")
+	}
+	if len > cap {
+		panic("reflect.MakeSlice: len > cap")
+	}
```

Comment:
You could consider calling `runtime.slicePanic` here, just like the compiler does. It should reduce binary size a bit at the cost of the readability of panic messages.